src/Facade/Services/FacadeResourceService2.cs

```diff
@@ -0,0 +1,194 @@
+            {
+                return new BadRequestResult<IEnumerable<TResource>>("Search is not implemented");
+            }
+        }
+
+        public IResult<TResource> Add(TResource resource, IEnumerable<string> privileges = null, int? userNumber = null)
+        {
+            T entity;
+
+            try
+            {
+                entity = this.CreateFromResource(resource);
+            }
+            catch (DomainException exception)
+            {
+                return new BadRequestResult<TResource>(exception.Message);
+            }
+
+            this.repository.Add(entity);
+            this.MaybeSaveLog("Create", userNumber, entity, resource, default);
+
+            this.transactionManager.Commit();
+
+            return new CreatedResult<TResource>(this.BuildResource(entity, privileges));
+        }
+
+        public IResult<TResource> Update(TKey id, TUpdateResource updateResource, IEnumerable<string> privileges = null, int? userNumber = null)
+        {
+            var entity = this.FindById(id);
+            if (entity == null)
+            {
+                return new NotFoundResult<TResource>();
+            }
+
+            try
+            {
+                this.UpdateFromResource(entity, updateResource);
+            }
+            catch (DomainException exception)
+            {
+                return new BadRequestResult<TResource>($"Error updating {id} - {exception.Message}");
+            }
+
+            this.MaybeSaveLog("Update", userNumber, entity, default, updateResource);
+            this.transactionManager.Commit();
+
+            return new SuccessResult<TResource>(this.BuildResource(entity, privileges));
+        }
+
+        public IResult<TResource> GetApplicationState(IEnumerable<string> privileges = null)
+        {
+            return new SuccessResult<TResource>(this.BuildResource(privileges));
+        }
+
+        public IResult<TResource> DeleteOrObsolete(TKey id, IEnumerable<string> privileges = null, int? userNumber = null)
+        {
+            var entity = this.FindById(id);
+            if (entity == null)
+            {
+                return new NotFoundResult<TResource>();
+            }
+
+            this.DeleteOrObsoleteResource(entity);
+            this.MaybeSaveLog("Delete", userNumber, entity, default, default);
+            this.transactionManager.Commit();
+
+            return new SuccessResult<TResource>(this.BuildResource(entity, privileges));
+        }
+
+        protected abstract T CreateFromResource(TResource resource);
+
+        protected abstract void UpdateFromResource(T entity, TUpdateResource updateResource);
+
+        protected abstract Expression<Func<T, bool>> SearchExpression(string searchTerm);
+
+        protected abstract void DeleteOrObsoleteResource(T entity);
+
+        protected abstract void SaveToLogTable(string actionType, int userNumber, T entity, TResource resource, TUpdateResource updateResource);
```

Comment:
I'm not sure if you guys have spoken about this before or if it wouldn't work, but depending how often logging happens would we ever make this (and possibly the delete too) virtual methods instead of abstract, so we don't have to override them with not implemented if we weren't using them in that facade service?

Reply:
that's not a bad idea. They're abstract to advertise that they're there really but perhaps the lesser used ones could be less in your face

Reply:
I prefer abstract fwiw. virtual would mean you'd have to provide a default implementation in the generic service which would probably lead you back towards the complications. Of course you could just throw a NotImplementedException in the default implemetation, but I think thats unclear as virtual implies 'I have a default implementation you can use'

Reply:
yeah. Also a good point. It's more dangerous having 'hidden' code that actually just throws an exception

Reply:
Yeah I did think it would be less obvious they were there, pretty much extreme laziness on my part thinking about writing a few lines less per facade 😄 Happy to leave as they are or up to you if you want to try one and see how we like it, thought it'd be worth asking and seeing others' opinions on it. I do get Lewis' point that it implies there's a default implementation, in practice I'd hope it's not too different from having our own not implemented(s) though - hopefully we'd just get an error and remember we have to implement bits